v2/charts/azure-service-operator/templates/apps_v1_deployment_azureserviceoperator-controller-manager.yaml

```diff
@@ -191,16 +199,14 @@ spec:
             path: /readyz
             port: 8081
           initialDelaySeconds: 60
+        {{- with .Values.resources }}
         resources:
-          limits:
-            cpu: 500m
-            memory: 512Mi
-          requests:
-            cpu: 200m
-            memory: 256Mi
+          {{- toYaml . | nindent 10 }}
+        {{- end }}
+        {{- with .Values.securityContext }}
```

Comment:
There's an issue related to this here: https://github.com/Azure/azure-service-operator/issues/4214

Can you expand what options you'd like to configure here? I'm somewhat of the opinion that we should be setting the most restrictive options we can here (I know we are not now), but I suppose opening it to allow users more control, as long as they're making things even more restrictive, makes some sense and is future-proofing in case we somehow miss something here.

Reply:
Sure, we're using options below - dropping all capabilities at container level and non root user ID at pod level probably due to presence of rbac proxy container in previous releases):
```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  ...
  name: azureserviceoperator-controller-manager
spec:
  ...
  template:
    ...
    spec:
      ...
      containers:
        - name: manager
          ...
          securityContext:
            capabilities:
              drop:
                - ALL
            readOnlyRootFilesystem: true
            allowPrivilegeEscalation: false
      ...
      securityContext:
        runAsUser: 10000
        runAsNonRoot: true
```

Reply:
We should probably make some of these the default, but will leave it for another PR.